src/cmakeProject.ts

```diff
@@ -389,7 +389,7 @@ export class CMakeProject {
             buildPreset,
             lightNormalizePath(this.folderPath || '.'),
             this.sourceDir,
-            this.workspaceContext.config.parallelJobs,
+            this.workspaceContext.config.isDefaultValue("parallelJobs") ? undefined : this.workspaceContext.config.parallelJobs,
```

Comment:
@sean-mcmanus Based on the `isDefaultValue` implementation, this seems to only check if the value is `undefined`. This makes me wonder if there's also a bug in the `isDefaultValue` implementation, shouldn't it be checking if the value equals the default value? 

As it stands, it seems like this would only ever return the `this.workspaceContext.config.parallelJobs`, unless the user explicitly set the value to `undefined`. 

Am I understanding this correctly?

Reply:
Another question, does this mean that in the presets case, we no longer use any default number of `parellelJobs`? This means that we won't parallelize at all in the default case? 

@bobbrow @iontodirel Any thoughts on this?

Reply:
@gcampbell-msft It is intentional that isDefaultValue is only checking for it being undefined and not equal to the actual default value. The reason is that undefined means the user has not set the value, while being 0 means they've actually set that.

In the preset case, no --parallel is set, but if the user is using Ninja, then it defaults to using "num logical cores" of parallelism, which is the same as if 0 were set for parallelJobs (the normal default as set by this PR).

Reply:
Gotcha, I think this makes sense then. I'll approve. 

It will go into pre-release in the next day and then the 1.21 official release when it comes.

Reply:
It feels to me like `ConfigurationReader.numJobs()` should be updated to check `isDefaultValue` so that we don't make this mistake again. And then we use `numJobs` everywhere instead of `parallelJobs`.  `parallelJobs` should also stop returning `| undefined` because it won't ever be `undefined`.

```ts
    get numJobs(): number | undefined {
        if (this.isDefaultValue("parallelJobs")) {
            return undefined;
        } else if (this.parallelJobs === 0) {
            return defaultNumJobs();
        } else {
            return this.parallelJobs;
        }
    }
```

Reply:
@bobbrow I think that's a good suggestion, @sean-mcmanus could you see if this works and try it out as a cleaner solution

Reply:
> @bobbrow I think that's a good suggestion, @sean-mcmanus could you see if this works and try it out as a cleaner solution

Yeah, because I think the current iteration can pass 0 for this argument which will be unexpected later on.

Reply:
@bobbrow @gcampbell-msft I didn't bother changing numJobs() because it's not used by the preset code and only the ctest code. But it sounds like you want me to update that?

Reply:
@bobbrow I don't know what you mean by parallelJobs not being undefined. I can still be.

Reply:
I can push an iteration with the changes I'm looking for if you want.

Reply:
Okay, yeah, I see it's 0 instead of undefined. I can change that.

Reply:
I think that it is from my perspective, I assume after the small refactors it still is working as expected for your project?

I think tracing through the code, it looks like everything is consistent with the original change, so I'm happy with it. 

I'll let @bobbrow review as well

Reply:
@gcampbell-msft Yes, it works as expected for me. My last push just fixed an issue/bug when parallelJobs was explicitly set to 0 -- I had removed the `parallelJobs || defaultNumJobs` at a spot, so not it'll be correctly set the defaultNumJobs.

Reply:
Looks good to me.